db/migrate/20210426151459_add_published_partial_index_to_podcasts.rb

```diff
@@ -0,0 +1,20 @@
+class AddPublishedPartialIndexToPodcasts < ActiveRecord::Migration[6.1]
+    disable_ddl_transaction!
+
+  def up
+    return if index_exists?(:podcasts, :published)
+
+    add_index :podcasts,
```

Comment:
I wonder if this index will be used at all. The table has only ~330 rows in total and for tables this small Postgres often opts for a FTS even when indexes are present.

Reply:
I had a similar thought, but I also wondered if the `LEFT OUTER JOIN` the search query generates would convince PostgreSQL to use it...

Reply:
I'm open to removing the extra indexes once we see what the production query planner does. How does that sound?

I was hoping to try a different approach where we throw more in at first, then remove the extra vs what we did on comments which resulted in 7982734892347 (read: 4) PRs 🤣

Reply:
@atsmith813 I just checked our [unused indexes for DEV](https://dev.to/admin/blazer/queries/455-unused-indexes) and eyeballing the numbers and, doing some napkin math, they take up less than 0.5% of our DB[1], so overindexing to be performance-cautious here is a great move. :100:

---

[1] This is a contender for the best value I've seen for this metric at any company I've worked at.

Reply:
@citizen428 

> I wonder if this index will be used at all. The table has only ~330 rows in total and for tables this small Postgres often opts for a FTS even when indexes are present.

IME, Postgres won't use an index for a query on a table with < 10k rows due to the cost of indexes on small data sets, however I agree with @rhymes that, due to the outer join, the table _projection_ may very well be over that 10k threshold. It'll be interesting to see.

Reply:
@jgaskins 10k feels about right 🙂 I guess I missed the outer join, that could well change it.

> [1] This is a contender for the best value I've seen for this metric at any company I've worked at.

We removed one of the bigger unused ones not too long ago: https://github.com/forem/forem/pull/11725